### PR TITLE
Per network services

### PR DESCRIPTION
**Pull request by elieux**
Please review, partially discussed on IRC.  Since there's a bunch of fixes, I recommend going commit by commit.

### CODE REVIEW

File: plugins/Services/plugin.py [hunk 1]
```diff
@@ -66,10 +66,16 @@ def __init__(self, irc):
...
-        self.waitingJoins = {}
+        self.state = {}
+
+    def _getState(self, irc):
+        class State(object):
+            def __init__(self):
+                self.channels = []
+                self.sentGhost = None
+                self.identified = False
+                self.waitingJoins = []
```
Comment (ProgVal) [inline plugins/Services/plugin.py:77]: Let's not create a class on every call... define it at the toplevel.
And no need to inherit from `object`

File: plugins/Admin/test.py [hunk 2]
```diff
@@ -31,13 +31,17 @@
...
     def testChannels(self):
         def getAfterJoinMessages():
             m = self.irc.takeMsg()
+            self.assertIsNotNone(m, 'No message back from joining channel.')
             self.assertEqual(m.command, 'MODE')
             m = self.irc.takeMsg()
+            self.assertIsNotNone(m, 'No message back from joining channel.')
             self.assertEqual(m.command, 'MODE')
             m = self.irc.takeMsg()
+            self.assertIsNotNone(m, 'No message back from joining channel.')
```
[the remark this answers is not in the document]
Comment (elieux) [reply plugins/Admin/test.py:44]: What about this then? https://github.com/ProgVal/Limnoria/blob/testing/src/test.py#L492-L499

File: plugins/Services/test.py [hunk 3]
```diff
@@ -105,6 +106,69 @@ def testRegisterNoExperimentalExtensions(self):
...
+            self.assertEquals(self.irc.takeMsg(), queuedJoin, 'Join request did not go through after identification.')
+        finally:
+            self.assertNotError('services password %s ""' % self.nick)
+
+    def testMultipleNetworks(self):
+        try:
+            net1 = copy(self)
+            net1.irc = getTestIrc('testnet1')
+            net1.assertNotError('services password %s %s' % (self.nick, 'secret'))
+            try:
```
Comment (ProgVal) [inline plugins/Services/test.py:152]: You can probably do this with a single `try` block

**Review by ProgVal — CHANGES_REQUESTED** [review]
`assertEqual` and 80 cols, please

File: plugins/Admin/test.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ProgVal) [reply plugins/Admin/test.py:44]: Uh, I don't know. But most plugins don't check it.

File: plugins/Services/plugin.py (hunk 1 above)
Comment (elieux) [reply plugins/Services/plugin.py:77]: Okay and okay.

File: plugins/Services/test.py (hunk 3 above)
Comment (elieux) [reply plugins/Services/test.py:152]: How to make sure both of the finally statements are attempted though?

File: plugins/Admin/test.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (elieux) [reply plugins/Admin/test.py]: Commit removed.

File: plugins/Services/plugin.py (hunk 1 above)
Comment (elieux) [reply plugins/Services/plugin.py]: Done and done.

File: plugins/Services/test.py (hunk 3 above)
Comment (elieux) [reply plugins/Services/test.py]: I guess the strictly correct behavior isn't worth the decrease in readability.  I merged the two blocks.

**Comment by elieux** [comment]
> `assertEqual`

I didn't know.  Changed now.

> 80 cols, please

Sorry.  I think I managed to fix all the long lines tastefully.

**Pull request merged**